File: c_files/raylib/extras/radar_sweep.c
#include "raylib.h"
#include <math.h>

int main(void) {
  const int screenWidth = 800;
  const int screenHeight = 600;
  const char screenTitle[] = "Radar Sweep";
  const Color screenBackground = BLACK;
  const int gameFps = 60;

  InitWindow(screenWidth, screenHeight, screenTitle);
  SetTargetFPS(gameFps);

  double angle = 0;

  double ring_x = (double)GetScreenWidth() / 2;
  double ring_y = (double)GetScreenHeight() / 2;
  double ring_inner_radius = 250;
  double ring_outer_radius = 260;
  double ring_start_angle = 0;
  double ring_end_angle = 360;
  double ring_segments = 50;
  Color ring_color = GREEN;

  double line_start_x = ring_x;
  double line_start_y = ring_y;
  double line_end_x = ring_x;
  double line_end_y = ring_y;
  double line_thickness = 5;
  Color line_color = YELLOW;

  while (!WindowShouldClose()) {

    angle++;
    if (angle > 360) {
      angle = 0;
    }

    double angle_radians = (PI / 180) * angle;

    line_end_x = ring_x + cos(angle_radians) * ring_inner_radius;
    line_end_y = ring_y + sin(angle_radians) * ring_inner_radius;

    BeginDrawing();
    ClearBackground(screenBackground);

    DrawRing((Vector2){ring_x, ring_y}, ring_inner_radius, ring_outer_radius,
             ring_start_angle, ring_end_angle, ring_segments, ring_color);

    DrawLineEx((Vector2){line_start_x, line_start_y},
               (Vector2){line_end_x, line_end_y}, line_thickness, line_color);

    EndDrawing();
  }

  CloseWindow();

  return 0;
}

File: python_files/raylib/extras/radar_sweep.py
import pyray as p
from math import radians, sin, cos

SCREEN_WIDTH: int = 800
SCREEN_HEIGHT: int = 800
SCREEN_TITLE: str = "Radar Sweep"
SCREEN_BACKGROUND: p.Color = p.SKYBLUE
GAME_FPS: int = 60


def main() -> None:
    p.init_window(SCREEN_WIDTH, SCREEN_HEIGHT, SCREEN_TITLE)
    p.set_target_fps(GAME_FPS)

    ring_center = p.Vector2(p.get_screen_width() / 2, p.get_screen_height() / 2)
    ring_inner_radius: float = 250
    ring_outer_radius: float = 260
    ring_start_angle: float = 0
    ring_end_angle: float = 360
    ring_segments: int = 64
    ring_color: p.Color = p.BLACK

    line_start_pos: p.Vector2 = ring_center
    line_thickness: float = 10
    line_color: p.Color = p.RED
    
    angle: float = 0

    while not p.window_should_close():

        line_end_pos = p.Vector2(
            ring_inner_radius * cos(radians(angle)) + ring_center.x,
            ring_inner_radius * sin(radians(angle)) + ring_center.y,
        )

        angle += 1
        if angle > 360:
            angle = 0
        p.begin_drawing()
        p.clear_background(SCREEN_BACKGROUND)

        p.draw_ring(
            ring_center,
            ring_inner_radius,
            ring_outer_radius,
            ring_start_angle,
            ring_end_angle,
            ring_segments,
            ring_color,
        )
        p.draw_line_ex(line_start_pos, line_end_pos, line_thickness, line_color)

        p.end_drawing()

    p.close_window()


if __name__ == "__main__":
    main()
